When we call `QueryCoord.Stop` before we initialize the `scheduler` of `QueryCoord`, `QueryCoord` will panic like below:

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x18313e2]



goroutine 1 [running]:

github.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).Close(0x0)

	/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:863 +0x22

github.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Stop(0xc0004f9600, 0x0, 0x0)

	/go/src/github.com/milvus-io/milvus/internal/querycoord/query_coord.go:226 +0x6c

github.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).Stop(0xc0001b70e0, 0xc0005c9b20, 0xc0005c9858)

	/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:275 +0x184

github.com/milvus-io/milvus/cmd/components.(*QueryCoord).Stop(0xc0001c60a0, 0xc0005d3bd0, 0x88b81b)

	/go/src/github.com/milvus-io/milvus/cmd/components/query_coord.go:59 +0x2f

github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc0005d3ef0, 0x0, 0x0, 0x0)

	/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:478 +0x5b3

main.main()

	/go/src/github.com/milvus-io/milvus/cmd/main.go:302 +0x635
```

--- a/internal/querycoord/task_scheduler.go
+++ b/internal/querycoord/task_scheduler.go
@@ -860,7 +860,9 @@ func (scheduler *TaskScheduler) Start() error {
 
 // Close function stops the scheduleLoop and the processActivateTaskLoop
 func (scheduler *TaskScheduler) Close() {
-	scheduler.cancel()
+	if scheduler.cancel != nil {
+		scheduler.cancel()
+	}
 	scheduler.wg.Wait()
 }
 

